include/cudaq/Optimizer/Dialect/CC/CCOps.td

```diff
@@ -388,6 +396,23 @@ def cc_IfOp : CCOp<"if",
         mlir::Location, mlir::Region &)>;
 
     bool hasResults() { return getResults().size(); }
+
+    // A cc.if must have a valid then region.
+    bool hasThen() { return !getThenRegion().empty(); }
+    mlir::Block *getThenEntryBlock() { return &getThenRegion().front(); }
+    mlir::Block::BlockArgListType getThenEntryArguments() {
+      return getThenEntryBlock()->getArguments();
+    }
```

Comment:
Will this handle a case like this:
```cpp
if (condition) {}
else {}
```
It seems like line 404 might segfault, but I might be missing something. (Lines 407-415 seem to handle this correctly/differently for the `else` block.)

Reply:
The `then` region is required to be non-empty. So it has a block and the block has a terminator. For the example given, a bridge could generate a similar `else` region as well or omit it.

At that point, it is up to the optimizer to discover, "hey, this if statement really doesn't do anything" and erase it.

A segfault isn't possible here because the block is required and a block has a list of arguments which is 0 or more BlockArguments in length.